### PR TITLE
Fixed PathView `MakeRelativeTo` and `Append` functions path segment comparisons 

### PR DESCRIPTION
**Pull request by lumberyard-employee-dm**
When using the Windows path separator of '\' the PathView class should be using case-insensitive comparisons of each path segment

The path segment comparisons were case-sensitive in both those functions
and now use `Internal::ComparePathSegments` function to perform the
appropriate case comparison based on the path separator value of the
Path class

Reverted the LocalFileIO::CheckInvalidWrite function back to not
lowercasing the assets alias and input path before invoking
`PathView::IsRelativeTo`

Signed-off-by: lumberyard-employee-dm <56135373+lumberyard-employee-dm@users.noreply.github.com>

### CODE REVIEW

**Review by lawsonamzn — APPROVED** [review]

**Review by hultonha — APPROVED** [review]
Thanks for fixing this! I believe this should resolve SPEC-8094, cheers!

**Review by amznestebanpapp — APPROVED** [review]

File: Code/Framework/AzCore/AzCore/IO/Path/PathParser.inl [hunk 1]
```diff
@@ -0,0 +1,706 @@
...
+    }
+
+    //! Check if a path is absolute on a OS basis
+    //! If the preferred separator is '/' just checks if the path starts with a '/
+    //! Otherwise a check for a Windows absolute path occurs
+    //! Windows absolute paths can include a RootName
+    template <typename InputIt, typename EndIt, typename = AZStd::enable_if_t<AZStd::Internal::is_input_iterator_v<InputIt>>>
+    static constexpr bool IsAbsolute(InputIt first, EndIt last, const char preferredSeparator)
+    {
+        size_t pathSize = AZStd::distance(first, last);
```
Comment (bosnichd) [inline Code/Framework/AzCore/AzCore/IO/Path/PathParser.inl:153]: When AZ_TRAIT_CUSTOM_PATH_ROOT_SEPARATOR is defined, this results in an unused local variable warning (that is treated as an error). In https://github.com/o3de/o3de/pull/3678 I just moved it to be calculated right before it's used in the two places below.

Comment (lumberyard-employee-dm) [reply Code/Framework/AzCore/AzCore/IO/Path/PathParser.inl:153]: It is not an unused variable, because the else block always uses it.

Comment (bosnichd) [reply Code/Framework/AzCore/AzCore/IO/Path/PathParser.inl:153]: Ah true, I hit that when I originally had the #if check at the function scope but the pathSize outside of it. Never mind me...

**Pull request merged**